src/SFA.DAS.LevyTransferMatching.Infrastructure/Services/OpportunitiesService/OpportunitiesService.cs

```diff
@@ -22,5 +23,23 @@ public async Task<List<OpportunityDto>> GetAllOpportunities()
 
             return JsonConvert.DeserializeObject<List<OpportunityDto>>(await response.Content.ReadAsStringAsync());
         }
+
+        public async Task<OpportunityDto> GetOpportunity(int id)
```

Comment:
I'm not 100% sure I created the DTO originally, actually. Nor did I create the `Dto` namespace/folder, just FYI.

Yes I re-used this model, as I have multiple times in previous projects (especially when one method returns a singular in a collection, and you have another method that returns a whole collection), and differ in opinion with regards to your own as to whether you should re-use models (generally, in my experience, re-use of existing models/code is seen as _good_, rather than a negative thing).

In the past, if models needed to change for different methods/uses, I've had no issues at all in modifying the models at the time, to suit the new approach, and/or creating inheriting types that contain more detial, if required. It can be quite elegant. But I certainly see your perspective on this (it's certainly keeping it simple and untangled), and am happy to follow this approach, even if it does mean lots of copy and paste. I was just unaware as to what extent you want to avoid model re-use.

Your comment on this probably also answers my questions previously, too.

Please could you confirm if you'd like me to shift the models into a 'Types' folder/namespace? I can do this, but it may make merging efforts for dependant branches awkward.

Reply:
Definitely shift the models that you are using/creating, but shifting the others is optional - do it if you want to, otherwise we can pick up as tech debt.

Reply:
I think we should probably review as a team (so everyone is on board) the plan for request/response models for the services, and review it as tech-debt. This means everyone will be on the same page and know what to expect when it comes to the inevitable merge.